api/views/team.py

```diff
@@ -0,0 +1,128 @@
+    def get_team_registration(self, request, pk=None):
+        
+        course = get_object_or_404(CanvasCourse, pk=pk)
+        userId = self.request.query_params.get('userId', None)
+        user = get_object_or_404(MyUser, pk=userId)
+        try:
+            team_reg = TeamRegistration.objects.get(user=user, team__course=course)
+        except ObjectDoesNotExist:
+            return Response({
+                'userId': userId,
+                'teamId': -1
+            })
+       
+        return Response({
+            'userId': team_reg.user.id,
+            'teamId': team_reg.team.id
+        })
+
+
+    @action(detail=True, methods=['post'])
+    def create_team(self, request, pk=None):
+        """
+        This endpoint completes team creation and registers the user creating the team in the course 
+        """
+
+        # get the team name from the request
+        team_name = request.data.get("name", None)
+
+        course = get_object_or_404(CanvasCourse, pk=pk)
+
+        # all teams registrations for the user in the requested course
+        all_team_reg = TeamRegistration.objects.filter(user=request.user, team__course=course)
+        
+        # delete other registrations if any exist
+        if(all_team_reg is not None):
+                for reg in all_team_reg:
+                    reg.delete()
+
+                    
+        # create a new team with the specified name
+        new_team = Team(name=team_name, course=course)
+        new_team.save()
+
+        # register the creator of the team to the team
+        new_team_reg = TeamRegistration(user=request.user, team=new_team)
+        new_team_reg.save()
+        
+
+        return Response({
+                "status": "Created",
+            })
+            
+    @action(detail=True, methods=['post'])
+    def join_team(self, request, pk=None):
+        """
+        This endpoint completes team registration for this (user, team) pair
+        """
+        team = get_object_or_404(Team, pk=pk)
+        team_reg = get_team_registration(user=request.user, team=team)
+
+        course = team.course
+        
+        # all teams registrations for the user in the requested course
+        all_team_reg = TeamRegistration.objects.filter(user=request.user, team__course=course)
+        
+
+        # if the user is not registered for the selected team
+        if team_reg is None:
+            # delete other registrations if any exist already
+            if(all_team_reg is not None):
+                for reg in all_team_reg:
+                    reg.delete()
+
+            # create a TeamRegistration for this (user, team) pair if one does not already exist
+            team_reg = TeamRegistration(user=request.user, team=team)
+            team_reg.save()
+
+        return Response({
+                "status": "Registered",
+            })
+        
+
+    @action(detail=True, methods=['post'])
+    def leave_team(self, request, pk=None):
```

Comment:
This could probably be a `delete` method. Then you can just override the `destroy` method and set the user's `team_reg` to `None` instead of completely deleting the object. Look at the `destroy` function in the question viewset. Changing this to override the `destroy` method means you could then make a `delete` request from the front-end.